pkg/ingestor/parser/scorecard/parser_scorecard.go

```diff
@@ -55,95 +54,68 @@ func (p *scorecardParser) Parse(ctx context.Context, doc *processor.Document) er
-// func (p *scorecardParser) CreateNodes(ctx context.Context) []assembler.GuacNode {
-// 	nodes := []assembler.GuacNode{}
-// 	for _, n := range p.scorecardNodes {
-// 		nodes = append(nodes, n)
-// 	}
-// 	for _, n := range p.artifactNodes {
-// 		nodes = append(nodes, n)
-// 	}
-//
-// 	return nodes
-// }
-//
-// // CreateEdges creates the GuacEdges that form the relationship for the graph inputs
-// func (p *scorecardParser) CreateEdges(ctx context.Context, foundIdentities []common.TrustInformation) []assembler.GuacEdge {
-// 	// TODO: handle identity for edges (https://github.com/guacsec/guac/issues/128)
-// 	edges := []assembler.GuacEdge{}
-// 	for i, s := range p.scorecardNodes {
-// 		edges = append(edges, assembler.MetadataForEdge{
-// 			MetadataNode: s,
-// 			ForArtifact:  p.artifactNodes[i],
-// 		})
-// 	}
-// 	return edges
-// }
-
-func (p *scorecardParser) GetPredicates(ctx context.Context) []assembler.PlaceholderStruct {
-	return nil
+func (p *scorecardParser) GetPredicates(ctx context.Context) *assembler.PlaceholderStruct {
+	var preds []assembler.CertifyScorecardIngest
+	for i, scPred := range p.scorecardPredicates {
+		preds = append(preds, assembler.CertifyScorecardIngest{
+			Scorecard: scPred,
+			Source:    p.srcPredicates[i],
+		})
+	}
+	return &assembler.PlaceholderStruct{
+		CertifyScorecard: preds,
+	}
 }
 
 // GetIdentities gets the identity node from the document if they exist
 func (p *scorecardParser) GetIdentities(ctx context.Context) []common.TrustInformation {
 	return nil
 }
 
-func metadataId(s *sc.JSONScorecardResultV2) string {
-	return fmt.Sprintf("%v:%v", s.Repo.Name, s.Repo.Commit)
+func (p *scorecardParser) GetIdentifiers(ctx context.Context) (*common.IdentifierStrings, error) {
+	return nil, fmt.Errorf("not yet implemented")
 }
 
-func getMetadataNode(s *sc.JSONScorecardResultV2) assembler.MetadataNode {
-	mnNode := assembler.MetadataNode{
-		MetadataType: "scorecard",
-		ID:           metadataId(s),
-		Details:      map[string]interface{}{},
+func getPredicates(s *sc.JSONScorecardResultV2) (*model.ScorecardInputSpec, *model.SourceInputSpec) {
+	var ns, name string
+	idx := strings.LastIndex(s.Repo.Name, "/")
+	if idx < 0 {
+		name = s.Repo.Name
 	}
 
-	for _, c := range s.Checks {
-		mnNode.Details[strings.ReplaceAll(c.Name, "-", "_")] = c.Score
-	}
-	mnNode.Details["repo"] = sourceUri(s.Repo.Name)
-	mnNode.Details["commit"] = hashToDigest(s.Repo.Commit)
-	mnNode.Details["scorecard_version"] = s.Scorecard.Version
-	mnNode.Details["scorecard_commit"] = hashToDigest(s.Scorecard.Commit)
-	mnNode.Details["score"] = float64(s.AggregateScore)
+	ns = s.Repo.Name[:idx]
+	name = s.Repo.Name[idx+1:]
 
-	return mnNode
-}
-
-func getArtifactNode(s *sc.JSONScorecardResultV2) assembler.ArtifactNode {
-	return assembler.ArtifactNode{
-		Name:   sourceUri(s.Repo.Name),
-		Digest: hashToDigest(s.Repo.Commit),
+	srcInput := model.SourceInputSpec{
+		// assuming scorecards is only git
+		Type:      "git",
```

Comment:
Can we get it from `s.Repo.Name`?

Reply:
hmm you mean like parsing and saying "github.com" and "gitlab.com" is git and "www.mercurial-scm.org" is "hg"?

Reply:
OH, I thought `s.Repo` contains a full URL like `git@github.com:guacsec/guac`, but if it is not then nothing to do